collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -121,32 +119,34 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
     }
 
     override fun getLists(): Set<String> {
-        return databaseConnection
-            .readableDatabase
-            .query(ListsTable.TABLE_NAME)
-            .foldAndClose(emptySet()) { set, cursor -> set + cursor.getString(ListsTable.COLUMN_NAME) }
+        return databaseConnection.withConnection {
+            readableDatabase
+                .query(ListsTable.TABLE_NAME)
+                .foldAndClose(emptySet()) { set, cursor -> set + cursor.getString(ListsTable.COLUMN_NAME) }
+        }
     }
 
     override fun updateListHash(list: String, hash: String) {
         val contentValues = ContentValues().also {
             it.put(ListsTable.COLUMN_HASH, hash)
         }
 
-        databaseConnection
-            .writeableDatabase
-            .update(
+        databaseConnection.withConnection {
+            writeableDatabase.update(
```

Comment:
I've just spotted that there is a typo here it should be `writableDatabase`.

Reply:
English is the worst!